docs/hazmat/primitives/asymmetric/dsa.rst

```diff
@@ -74,8 +74,55 @@ provider.
     ...     key_size=1024,
     ...     backend=default_backend()
     ... )
-    >>> signer = private_key.signer(hashes.SHA256())
     >>> data = b"this is some data I'd like to sign"
+
+Signing using a :class:`cryptography.hazmat.primitives.hashes.HashContext` (preferred method):
+
+.. doctest::
+
+    >>> digester = hashes.Hash(hashes.SHA256(), backend=default_backend())
+    >>> digester.update(data)
+    >>> signature = private_key.sign(digester)
+
+One-line signing:
+
+.. doctest::
+
+    >>> signature = private_key.sign(data, hashes.SHA256())
+
+Signing a precomputed digest:
+
+.. warning::
+
+    Only use this for digests, do NOT sign raw data. The data will get truncated
+    to the size of DSA's "q" parameter, which is just a few bytes.
+
+.. doctest::
+
+    >>> digester = hashes.Hash(hashes.SHA256(), backend=default_backend())
+    >>> digester.update(data)
+    >>> digest = digester.finalize()
+    >>> signature = private_key.sign(digest, already_hashed=True)
+
+Specifying the per-message key k:
```

Comment:
Under what scenario do you want the ability to set `k` manually? I realize there's a warning below, but this API is incredibly dangerous.